ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/handler/TestAbortedTxnCleaner.java

```diff
@@ -282,9 +284,19 @@ public void testCleaningOfAbortedDirectoriesBelowBase() throws Exception {
     cleaner.setCleanupHandlers(Arrays.asList(mockedTaskHandler));
     cleaner.run();
 
-    Mockito.verify(mockedFSRemover, Mockito.times(1)).clean(any(CleanupRequest.class));
+    Mockito.verifyNoInteractions(mockedFSRemover);
```

Comment:
Why this change in test? If the cleaner has run then ideally to delete the files there is call made to FsRemover mocked object. Also there is only one compaction request on this table, so this request cannot be skipped.

Reply:
The test creates 4 deltas on the same table and the same partition: 1 committed, 2 aborted and lastly 1 committed. 

When the cleaner runs there, it cannot do the clean up because there is a newer record in the compaction queue with higher write id for the given table and partition (as the aborted ones are in the middle of the set up). 

So that they won't be cleaned, they delete nothing, as the ticket requests it. 

At the end of the test, we run the whole compaction flow again and finish all the transactions. So that, the directories will be cleaned up. 

In my opinion, that is the exact behaviour that was requested in the ticket. 

What do you think about that?

Reply:
@InvisibleProgrammer Thanks for the explanation. Can we assert the fact that there is a compaction record in the test?

Reply:
Added an assert on compaction_queue.